### PR TITLE
[MOSIP-34073] created terraform scripts & workflows

### PR DESCRIPTION
**Pull request by syedsalman3753**

### CODE REVIEW

File: terraform/aws/README.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+## Destroy MOSIP Infrastructure
+
+### Run `terraform destroy` workflow 
+* This GitHub Action automates the `Terraform destroy` command within a CI/CD pipeline.
+The workflow can be manually triggered to destroy infrastructure managed by Terraform.
+* To trigger this workflow, go to the `Actions` tab in your GitHub repository. Select `terraform destroy` workflow and provide the required inputs and click `Run workflow` to start the workflow.
+  ![terraform-destroy-1.png](../../docs/_images/terraform-destroy-1.png)
+
+### Inputs
+* `SSH_PRIVATE_KEY (required)`: GitHub secret name containing the private key for SSH login on the nginx node.
```
Comment (ckm007) [inline terraform/aws/README.md:34]: Not needed.

File: terraform/aws/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+* `TERRAFORM_DESTROY (optional)`: Boolean flag to determine whether to execute the Terraform destroy command. Defaults to false.
+
+### Environment Variables
+* `AWS_ACCESS_KEY_ID`: The AWS access key ID used for authenticating with AWS services. This is stored as a GitHub secret `AWS_ACCESS_KEY_ID`.
+* `AWS_SECRET_ACCESS_KEY`: The AWS secret access key corresponding to the AWS access key ID. This is stored as a GitHub secret `AWS_SECRET_ACCESS_KEY`.
+* `TF_VAR_SSH_PRIVATE_KEY`: The SSH private key used for accessing nginx server. This is referenced from the input `SSH_PRIVATE_KEY` and stored as a GitHub secret.
```
Comment (ckm007) [inline terraform/aws/README.md:40]: not needed.

File: terraform/aws/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+# Terraform Setup for MOSIP Infrastructure
+
+## Overview
+This Terraform configuration script set up the infrastructure for MOSIP (Modular Open Source Identity Platform) on AWS.
```
Comment (ckm007) [inline terraform/aws/README.md:45]: MOSIP (Modular Open Source Identity Platform) on AWS --> MOSIP deployment on AWS.

File: terraform/aws/README.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+The setup includes security groups, an NGINX server, and a Kubernetes (K8S) cluster.
```
Comment (ckm007) [inline terraform/aws/README.md:46]: The setup includes security groups, an NGINX server, and a Kubernetes (K8S) cluster.

-->

The setup includes:
1. EC2 instance creation for Kubernetes nodes.
2. Security group defination for cluster nodes.
3. EC2 instance for Nginx server.
4. Security group for Nginx machine.

Comment (ckm007) [reply terraform/aws/README.md:46]: etc

File: terraform/aws/modules/nginx-setup/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+## Terraform & Shell Script for Nginx Setup with SSL
+
+## Overview
+This Terraform configuration script sets up a Nginx server with SSL certificates on an AWS EC2 instance.
+It fetches SSL certificates using Certbot and integrates with Kubernetes infrastructure from a specified GitHub repository
```
Comment (ckm007) [inline terraform/aws/modules/nginx-setup/README.md:5]: also lets add the cron addition for renewal here.

**Review by ckm007 — CHANGES_REQUESTED** [review]

File: terraform/aws/modules/nginx-setup/README.md (hunk 5 above)
Comment (syedsalman3753) [reply terraform/aws/modules/nginx-setup/README.md:5]: Have requested for few changes to @Mohanraj209, will take it in an another PR

File: .github/workflows/terraform-destroy.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+name: terraform destroy
+
+on:
+  workflow_dispatch:
+    inputs:
+      SSH_PRIVATE_KEY:
```
[the remark this answers is not in the document]
Comment (syedsalman3753) [reply .github/workflows/terraform-destroy.yml:6]: Yes, as Terraform doesn't store all the input variables in .tfstate file. 
Therefore, passing during terraform plan, apply, or destroy is required.

File: .github/workflows/terraform-destroy.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        type: boolean
+        default: false
+
+env:
+  # TF_LOG_PATH: ./temp/terraform.log
+  ## TRACE, DEBUG, INFO, WARN or ERROR
+  # TF_LOG: TRACE
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  TF_VAR_SSH_PRIVATE_KEY: ${{ secrets[inputs.SSH_PRIVATE_KEY] }}
```
[the remark this answers is not in the document]
Comment (syedsalman3753) [reply .github/workflows/terraform-destroy.yml:21]: Yes, as Terraform doesn't store all the input variables in .tfstate file.
Therefore, passing during terraform plan, apply, or destroy is required.

File: terraform/aws/README.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
+# Terraform Setup for MOSIP Infrastructure via GitHub Actions
+
+## Create MOSIP Infrastructure 
+
+### Prerequisites
+* Create a new branch `env-<environment_name>` from master/develop branch.
+* Goto `terraform/aws` location and update environment related details in `env.tfvars` file.
```
[the remark this answers is not in the document]
Comment (syedsalman3753) [reply terraform/aws/README.md:7]: The `env.tfvars` file includes only the necessary variables to provide values. The input variables referenced in the README files have additional requirements that will be handled by the Terraform script. Therefore, there is no need to refer to the README files for this purpose.

[the remark this answers is not in the document]
Comment (ckm007) [reply terraform/aws/README.md:7]: we need to make sure that user understands the necesary variables so comments are needed.
I currently see that we are adding what is variable is in values field which is again not correct. So please correct it.

File: terraform/aws/README.md (hunk 3 above)
Comment (ckm007) [reply terraform/aws/README.md:45]: I dont see this is taken care. please correct.

**Review by ckm007 — APPROVED** [review]

**Pull request merged**